### ISSUE TITLE
"Expected 0 positional arguments" with ParamSpec and non-generic return types

### ISSUE THREAD

**Issue by Terrance** [microsoft/pyright#1867]
## Describe the bug

For a callable of type `Callable[P, T]` where `P` is a `ParamSpec`, Pyright states no arguments are accepted​, but only when `T` is a non-generic type.

## To reproduce

```python
from typing import Callable, ParamSpec, TypeVar

T = TypeVar("T")
P = ParamSpec("P")
```

This works fine (as does e.g. a subclass of `Generic[T]` in place of `T`):

```python
def deco(fn: Callable[P, T]) -> Callable[P, T]:
    def inner(*args: P.args, **kwargs: P.kwargs) -> T:
        return fn(*args, **kwargs)
    return inner

@deco
def handle(x: int):
    print(x)
    return x

handle(1)
```

But if we swap `T` for a concrete type...

```python
def deco_int(fn: Callable[P, int]) -> Callable[P, int]:
    def inner(*args: P.args, **kwargs: P.kwargs) -> int:
        return fn(*args, **kwargs)
    return inner

@deco_int
def handle_int(x: int):
    print(x)
    return x

handle_int(1)  # Expected 0 positional arguments
handle_int(a=1)  # No parameter named "a"

handle_int()  # (type-valid, but a runtime error)
```

![image](https://user-images.githubusercontent.com/4025899/118394409-7c9eb180-b63c-11eb-813f-1552aee96fad.png)

## VS Code extension or command-line

Pyright 1.1.139 in VSCode.

## Additional context

I'm using Python 3.10.0b1 to test -- version 3.10.0.0 of typing_extensions was [released](https://github.com/python/typing/releases/tag/3.10.0.0) a couple of weeks ago, but Pyright isn't currently happy with me using it:

```python
from typing_extensions import ParamSpec

P = ParamSpec("P")  # ParamSpec requires Python 3.10 or newer
```

The Python version check should be relaxed to Python 3.4+ when accessed via extensions.

**Comment by erictraut** [comment]
Thanks for the bug report. I have dozens of unit tests for ParamSpec usage, but all of them used generics in the return type, so this bug went uncaught.

This will be fixed in the next release.

**Comment by Terrance** [comment]
Great, thanks.  Any update on `typing_extensions.ParamSpec` in older Python versions?  Doesn't look like I can suppress errors for that in the meantime.

**Comment by erictraut** [comment]
I'm pretty sure that the latest version of `typing_extensions` contains support for `ParamSpec`, but you could check with the maintainers of that library for confirmation.

**Comment by Terrance** [comment]
It's in there and valid at runtime, but Pyright is flagging it (see additional context above).

**Comment by erictraut** [comment]
Ah, I see what you mean. Yes, pyright shouldn't be flagging this as an error. This will be fixed as well.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.140, which I just published. It will also be in the next release of pylance.
